scripts/clean_osm_data.py

```diff
@@ -597,11 +716,11 @@ def prepare_generators_df(df_all_generators):
 
 
 def find_first_overlap(geom, country_geoms, default_name):
-    """Return the first country whose shape intersects the geometry"""
-    for c_name, c_geom in country_geoms.items():
-        if not geom.disjoint(c_geom):
-            return c_name
-    return default_name
+    """Return the first index whose shape intersects the geometry"""
+    return next(
+        (idx for idx, c_geom in country_geoms.items() if not geom.disjoint(c_geom)),
+        default_name,
+    )
```

Comment:
Not sure it became more clear :( Indeed, an initial nested loop isn't the most elegant approach. But I fear re-formulating if as a nested loop in functional programming style might be a bit harsh towards a code reader

Wouldn't it be more clear to keep one explicit loop?

Reply:
This approach was proposed to increase the efficiency of the code and reduce the overhead.
The function next exactly replace the previous approach exploiting native python activities that are expected to be highly efficient.

However, after this check: https://realpython.com/python-first-match/
The increase in performances is negligible. I revert it back.
Thanks!

Reply:
Thanks a lot for the explanation!